Include bind arguments as part of equals/hashCode in BindConfiguration

hjohn opened the pull request:
We discovered that when creating two queue bindings on the same exchange which only differ in binding arguments that rabbit mock will not deliver messages to the queue for all of the possible bindings (ie, one of the bindings simply does not work despite a message being delivered to the exchange with correct arguments).

This is because the BindConfigurations are at some point compared and its equals/hashCode does not take the binding arguments into account.

This PR shows a potential solution which fixes our test cases.

ledoyen commented:
Hello @hjohn and thank you for you contribution.

Could you add a test that asserts that this fix allows multiple bindings ?

Thanks in advance,

hjohn commented:
> Could you add a test that asserts that this fix allows multiple bindings ?

I've added a test, and also fixed the unbinding -- unbind requires the use of the arguments originally passed to bind to be able to remove the exact binding created (including the headers).

The test case I added tests that messages are received correctly and that unbinding has the desired effect.

ledoyen commented:
Thanks for the change, could you please squash the two commits ?

hjohn commented:
> Thanks for the change, could you please squash the two commits ?

Should be good now
